modules/dataframe/src/datastructures/column.cpp

```diff
@@ -72,33 +119,68 @@ CategoricalColumn* CategoricalColumn::clone(const std::vector<std::uint32_t>& ro
 
 ColumnType CategoricalColumn::getColumnType() const { return ColumnType::Categorical; }
 
-std::string CategoricalColumn::getAsString(size_t idx) const {
-    auto index = getTypedBuffer()->getRAMRepresentation()->getDataContainer()[idx];
-    return lookUpTable_[index];
+const std::string& CategoricalColumn::getHeader() const { return header_; }
+
+void CategoricalColumn::setHeader(std::string_view header) { header_ = header; }
+
+Unit CategoricalColumn::getUnit() const { return unit_; }
+
+void CategoricalColumn::setUnit(Unit unit) { unit_ = unit; }
+
+void CategoricalColumn::setCustomRange(std::optional<dvec2> range) { range_ = range; }
+
+std::optional<dvec2> CategoricalColumn::getCustomRange() const { return range_; }
+
+dvec2 CategoricalColumn::getDataRange() const {
```

Comment:
Should not, but can due to being able to register categories and edit the buffer.

Reply:
is this something that we should treat as an invariant? can there be unused labels in the map? are you allowed to change the buffer directly? to we validate that It any change to the buffer still points to a valid label?

Reply:
Yes, there might be unused labels caused by either registering a new category or when filtering the column and removing a subset. I would argue against compacting those as it might interfere with color mapping for example.
Editing the buffer gives you a bit more freedom than using `addMany()`. There is however no validation at the moment. Might add one which throws if there is a problem (or alternatively needs some heuristics to fix those by introducing dummy strings).